### ISSUE TITLE
There is a bug at 'dataset.py'

### ISSUE THREAD

**Issue by CGLemon** [yukw777/leela-zero-pytorch#3]
Stone places will be reading in container by hex. But hex format of numpy is different with format of Leelaz.
https://github.com/yukw777/leela-zero-pytorch/blob/72024b36aaf0fb0853cb7d6ce3ab8cd17f6adc12/leela_zero_pytorch/dataset.py#L35
https://github.com/yukw777/leela-zero-pytorch/blob/72024b36aaf0fb0853cb7d6ce3ab8cd17f6adc12/leela_zero_pytorch/dataset.py#L19

The stone places format is defined here.
https://github.com/leela-zero/leela-zero/blob/b259e50d5cce34a12176846534f369ef5ffcebc1/src/Training.cpp#L245

**Comment by yukw777** [comment]
i don't quite get what you mean when you say the hex formats of numpy and LeelaZero are different. Could you be more specific? Is padding scheme, or maybe endianness different? What is the wrong behavior caused by this bug?

I referenced the tensorflow training pipeline (https://github.com/leela-zero/leela-zero/blob/b259e50d5cce34a12176846534f369ef5ffcebc1/training/tf/chunkparser.py#L160-L222), and wrote some unit tests to make sure I'm parsing the move format correctly too (https://github.com/yukw777/leela-zero-pytorch/blob/72024b36aaf0fb0853cb7d6ce3ab8cd17f6adc12/leela_zero_pytorch/test_dataset.py#L10-L58), so maybe you can take a look at that and help me see where the bug is.

**Comment by CGLemon** [comment]
Because there is same issue at board size 3 , lets set the board size to 3 to make the issue easier.

There is a black stone place in Leelaz, like this.
[   1    0    1
    0    0    0
    0    0    1  ]

The first to 8th were transformed to two bits, 'a' and '0'. And the last one is defferent. If it is zero , you will get zero. If it is one , you will get one. The output in the gzip file is 'a01'

In the 'dataset.py' , we will get stone place is 
[   0    0    0
    0    0    0
    0    0    1 ]

Becase 'dataset.py' shift right 7 bits, the endianness format is different with Leelaz.

**Comment by yukw777** [comment]
@CGLemon oh my god, great catch!! I have the fix in https://github.com/yukw777/leela-zero-pytorch/tree/stone-plane-bug. I'm retraining the small network right now to verify everything works as expected. Once things look good, I'll merge it to master! Thanks again for catching this!!!! Please let me know if you see any other issues.
